### ISSUE TITLE
Merge conflict warning when there are none

### ISSUE THREAD

**Issue by ben-clayton** [Microsoft/vscode#44106]
### Issue Type
Bug

### Description

1. Rebase a Git branch on to another containing identical changes (but without the abiltiy for git to fast-forward).
2. Use VS code to resolve the conflicts so that a file no longer contains any diffs.
3. Observe that there are no conflict markers remaining in the file.
3. Attempt to stage the file.
4. Observe the `Are you sure you want to stage XXX with merge conflicts?` message.
5. Click ok.
6. Observe the file disappears from the source control list (as expected due to the no-delta change).

(5) Is unexpected. 

I've also checked that the files do not contain any `<<<`, `===`, `>>>` looking comments.

This happens more often than not, but I have seen it stage the file without warning before. I was unable to determine what was different with this situation.

Observed on Linux, macOS and Windows.

### VS Code Info

VS Code version: Code 1.20.1 (f88bbf9137d24d36d968ea6b2911786bfe103002, 2018-02-13T15:31:21.019Z)
OS version: Linux x64 4.13.0-32-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4930K CPU @ 3.40GHz (12 x 3400)|
|Load (avg)|1, 2, 1|
|Memory (System)|15.60GB (9.40GB free)|
|Process Argv|/usr/share/code/code --disable-extensions|
|Screen Reader|no|
|VM|0%|

</details>Extensions disabled
Reproduces without extensions

**Comment by ben-clayton** [comment]
To easily reproduce step 1 (tested on linux):

```
mkdir vscode-bug-repro
cd vscode-bug-repro

git init
git checkout -b branch-a
echo "Keep this line" > file.txt
git add file.txt
git commit -m "Stuff"

git checkout -b branch-b
echo "Delete this line" > file.txt 
git add file.txt
git commit --amend -m "Skip this change in merge"
git rebase branch-a

code .
```

For (2), resolve the merge conflicts to that `file.txt` just contains the single line `Keep this line`.

**Comment by dkeulen** [comment]
I also have the same issue unfortunately.
latest vscode (1.21.1 - 79b4.....20f1), windows 10.

**Comment by johannao76** [comment]
Same here. When trying to stage files, VSCode claims there are conflicts when there are none.
Version 1.22.0
Commit 6b4d53cdab8bcae1eaaa4934d93c077319b573db
Date 2018-04-05T17:27:42.363Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
OS Windows 10

**Comment by zhuowei** [comment]
I'm looking into this issue: here's what I found so far.

The code that displays the error is in

https://github.com/Microsoft/vscode/blob/d1380912cf5c9db73d35ba00c5028f021a8f897e/extensions/git/src/commands.ts#L636

I tested using the [repro from the first comment above](https://github.com/Microsoft/vscode/issues/44106#issuecomment-367292376), and after executing, `unresolvedConflicts` contains one item: the text file with status `Status.BOTH_ADDED`. However, the code seems to only check whether files with the `Status.BOTH_MODIFIED` status had conflicts resolved.

I'm guessing that the fix would involve checking if files with Status.BOTH_ADDED also had their conflicts resolved.
